test/evergreen/code_complexity_analysis.py

```diff
@@ -0,0 +1,137 @@
+    complexityDict['Top ' + str(topNRegions) + ' Regions'] = get_region_list(dataFile)
+    
+    resultList.append(complexityDict)
+    return (resultList)
+
+def get_region_list(dataFile):
+    ''' Retrieve topN Regions/functions with the highest cyclomatic complexity values. '''
+
+    top5 = dataFrame.nlargest(topNRegions, 'std.code.complexity:cyclomatic')
+    atlasFormat = {}
+    for index, row in top5.iterrows():
+        atlasFormat[row["region"]] = row["std.code.complexity:cyclomatic"]
+
+    return (atlasFormat)
+
+def get_complexity_ranges_list(dataFile):
+    ''' Retrieve 3 set of cyclomatic complexity ranges, above 20, above 50 and above 90. '''
+
+    rangesList = [20, 50, 90]
+    atlasFormat = {}
+    for i in rangesList:
+        aboveRangeString = 'Above ' + str(i)
+        atlasFormat[aboveRangeString] = get_complexity_max_limit(i)
+
+    return (atlasFormat)
+
+def get_complexity_max_limit(max_limit: int):
+    ''' Retrieve the number of cyclomatic complexity values above the max_limit. '''
+
+    columnName = 'std.code.complexity:cyclomatic'
+    # Select column 'std.code.complexity:cyclomatic' from the dataframe
+    column = dataFrame[columnName]
+    # Get count of values greater than max_limit in the column 'std.code.complexity:cyclomatic'
+    count = column[column > max_limit].count()
+
+    # Pandas aggregation functions (like sum, count and mean) returns a NumPy int64 type number not a Python integer.
+    # Object of type int64 is not JSON serializable so need to convert into an int.
+
+    return (int(count))
+
+def get_average(complexity_summary_file):
+    ''' Retrieve the cyclomatic code complexity value '''
+    """
+    Below is the format of complexity_summary_file, we would extract the "avg" value of std.code.complexity.
+    {
+        "view": [
+            {
+                "data": {
+                    "info": {"path": "./", "id": 1},
+                    "aggregated-data": {
+                        "std.code.complexity": {
+                            "cyclomatic": {
+                                "max": 91,
+                                "min": 0,
+                                "avg": 4.942372881355932,
+                                "total": 14580.0,
+                                "count": 2950,
+                                "nonzero": False,
+                                "distribution-bars": [ ],
+                                "sup": 0,
+                            }
+                        },
+                        "std.code.lines": {
+                            "code": { }
+                        },
+                    },
+                    "file-data": {},
+                    "subdirs": [ ],
+                    "subfiles": [ ],
+                }
+            }
+        ]
+    }
+    """
+
+    with open(complexity_summary_file, 'r') as f:
+        data = f.read()
+
+    d = ast.literal_eval(data)
+    newDictionary = d["view"]
```

Comment:
I think this line should be put into the try/except block as it's susceptible to hit an exception if the data is not formatted as expected.

Reply:
Fixed in 19d2a584b833bc89692c079e17033ea0c4609175